src/inputs/internal/SelectFieldBase.tsx

```diff
@@ -45,6 +45,8 @@ export interface BeamSelectFieldBaseProps<O, V extends Value> extends BeamFocusa
   contrast?: boolean;
   /** Placeholder content */
   placeholder?: string;
+  /** Only used for Single Select Fields. If set, prepends an option with a `undefined` value at the top of the list */
+  unsetLabel?: string;
```

Comment:
Question: Currently our `onSelect` has `values: V[]`...if a user uses `unsetLabel`, what do we call `onSelect` with? It looks like `undefined`... should we update `onSelect` to have `values: V[] | undefined`? 

Or should it be an empty list?

Fwiw getting this typing right, i.e. if you want an `unsetLabel` your V needs to `extends undefined` _but_ if you don't set `unsetLabel`, then :shrug: you don't have to accept `undefined`, has been primarily why we haven't done this before imo/iirc.

Reply:
We'd currently call `onSelect` with `values = [undefined]`

Reply:
I'm not recalling our whole `keyToValue`/`valueToKey` stuff... will see how I can better work within those bounds.

Reply:
Okay, great. So this whole rewriting of the `getOptionValue` is unnecessary because we do the `keyToValue` and `valueToKey` conversions in this component. Only thing we need to overwrite is the label helpers.

Reply:
...I only recall that those functions exist :-D will have to go refresh my memory.

Reply:
Hm, I'm not quite following...

* As a caller, I pass `options=[{id:1}, {id:2}, {id:3}]`
* I have `getOptionValue=o => o.id`
* I pass `unsetLabel=None`
* We inject `options=[unsetOption, {id:1}, {id:2}, ...]`

We're going to call `getOptionValue({})` and the caller is going to do `{}.id`...

I guess technically this will work for probably 90% of our `getOptionValue`s, but if a `getOptionValue` was `o => o.id.toString()`, that would compile as valid TS, but then fail at runtime...

So, I dunno, I think we still want to intercept `getOptionValue` and not call the user's `getOptionValue` with an empty `{}` that they are not suspecting.

Reply:
Great point. I will re-introduce overwriting the `getOptionValue` locally as well. Thanks for the thorough and thoughtful feedback.